### PR TITLE
board specific driver and module organization

### PR DESCRIPTION
**Pull request by dagar**
With the new board directories we now have a place where board specific drivers and modules can reside.

These components are automatically included via standard cmake mechanisms in the board itself. It further simplifies the board configuration as well as the core Firmware source tree.

### CODE REVIEW

**Comment by LorenzMeier** [comment]
Not building yet

**Comment by dagar** [comment]
Good to go.

File: boards/beaglebone/blue/bbblue_adc/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+	SRCS
+		bbblue_adc.cpp
+		
+		# Previous board specific code is now precluded from build after
+		# the following was added to Frimware/CMakeLists.txt:
+		#
+		# 		add_subdirectory(src/drivers/boards EXCLUDE_FROM_ALL)
+		#
+		# so include bbblue board specific code here:
+		../src/init.c
```
Comment (bkueng) [inline boards/beaglebone/blue/bbblue_adc/CMakeLists.txt]: Required?

Comment (dagar) [reply boards/beaglebone/blue/bbblue_adc/CMakeLists.txt]: Probably not, but I didn't want to get too far into it without this board even compiling in CI.

**Pull request merged**